src/Connector.php

```diff
@@ -42,24 +37,29 @@ public function __invoke($url, array $subProtocols = [], array $headers = []) {
         } catch (\Exception $e) {
             return new RejectedPromise($e);
         }
-        $connector = 'wss' === substr($url, 0, 3) ? $this->_secureConnector : $this->_connector;
+        $secure = 'wss' === substr($url, 0, 3);
+        $connector = $this->_connector;
 
-        $port = $uri->getPort() ?: 80;
+        $port = $uri->getPort() ?: ($secure ? 443 : 80);
 
-        return $connector->create($uri->getHost(), $port)->then(function(DuplexStreamInterface $stream) use ($request, $subProtocols) {
+        $uriString = $uri->getHost() . ':' . $port;
```

Comment:
What if the URL was something like `wss://socketo.me:9001`?

Perhaps if `$secure` we set the scheme to `tls://`?

Reply:
This is fixed in the latest commit.